momento-sdk/src/main/java/momento/sdk/batchutils/MomentoBatchUtils.java

```diff
@@ -0,0 +1,229 @@
+    return new MomentoBatchUtilsBuilder(cacheClient);
+  }
+
+  /**
+   * Performs a batch get operation for String keys.
+   *
+   * @param cacheName The name of the cache.
+   * @param request The batch get request with String keys.
+   * @return BatchGetResponse The batch get response
+   */
+  public CompletableFuture<BatchGetResponse> batchGet(
+      final String cacheName, final BatchGetRequest.StringKeyBatchGetRequest request) {
+
+    final List<CompletableFuture<BatchGetResponse.StringKeyBatchGetSummary.GetSummary>>
+        futureSummaries = new ArrayList<>();
+
+    for (final String key : request.getKeys()) {
+      final CompletableFuture<BatchGetResponse.StringKeyBatchGetSummary.GetSummary> futureSummary =
+          CompletableFuture.supplyAsync(
+              () -> {
+                try {
+                  final GetResponse getResponse =
+                      cacheClient
+                          .get(cacheName, key)
+                          .get(this.requestTimeoutSeconds, TimeUnit.SECONDS);
+                  return new BatchGetResponse.StringKeyBatchGetSummary.GetSummary(key, getResponse);
+                } catch (Exception e) {
+                  return new BatchGetResponse.StringKeyBatchGetSummary.GetSummary(
+                      key,
+                      new GetResponse.Error(CacheServiceExceptionMapper.convert(e.getCause())));
+                }
+              },
+              executorService);
+
+      futureSummaries.add(futureSummary);
+    }
+
+    // chain all the futures to generate a new future returned back to the caller
+    return CompletableFuture.allOf(futureSummaries.toArray(new CompletableFuture[0]))
+        .thenApply(
+            v -> {
+              List<BatchGetResponse.StringKeyBatchGetSummary.GetSummary> summaries =
+                  futureSummaries.stream()
+                      .map(CompletableFuture::join)
+                      .collect(Collectors.toList());
+              return new BatchGetResponse.StringKeyBatchGetSummary(summaries);
+            });
+  }
+
+  /**
+   * Performs a batch get operation for byte array keys.
+   *
+   * @param cacheName The name of the cache.
+   * @param request The batch get request with byte array keys.
+   * @return BatchGetResponse The batch get response
+   */
+  public CompletableFuture<BatchGetResponse> batchGet(
+      final String cacheName, final BatchGetRequest.ByteArrayKeyBatchGetRequest request) {
+
+    final List<CompletableFuture<BatchGetResponse.ByteArrayKeyBatchGetSummary.GetSummary>>
+        futureSummaries = new ArrayList<>();
+
+    for (final byte[] key : request.getKeys()) {
+      final CompletableFuture<BatchGetResponse.ByteArrayKeyBatchGetSummary.GetSummary>
+          futureSummary =
+              CompletableFuture.supplyAsync(
+                  () -> {
+                    try {
+                      GetResponse getResponse =
+                          cacheClient
+                              .get(cacheName, key)
```

Comment:
This call is an hindrance for me to refactor common logic out as the SDK doesn't implement Java generics. So if I add a generic wrapper for my BatchGetRequest, we do not know here the type.